File: frontend/src/types/webview.ts
export interface WebViewFinding {
  rule_id: string
  severity: string
  title: string
  file: string
  line: number
  evidence: string
}

export interface WebViewFile {
  index: number
  source: string
  path: string
  size_bytes: number
  findings: WebViewFinding[]
  bridge_methods: string[]
}

export interface WebViewScanResult {
  analysis_id: number
  files_found: number
  findings_count: number
  files: WebViewFile[]
}

File: backend/schemas/webview.py
from pydantic import BaseModel


class WebViewFinding(BaseModel):
    rule_id: str
    severity: str
    title: str
    file: str
    line: int
    evidence: str


class WebViewFile(BaseModel):
    index: int
    source: str
    path: str
    size_bytes: int
    findings: list[WebViewFinding]
    bridge_methods: list[str]


class WebViewScanResult(BaseModel):
    analysis_id: int
    files_found: int
    findings_count: int
    files: list[WebViewFile]
